Make CMake sanitizers optional, and opt-in

I am trying to build libappimage using as many components from the distro as possible (since libappimage might eventually become distro-packaged one day).

Following https://github.com/AppImage/AppImageUpdate/blob/rewrite/BUILDING.md, I get when building AppImageKit:

```
CMake Warning at src/CMakeLists.txt:1 (find_package):
  By not providing "FindSanitizers.cmake" in CMAKE_MODULE_PATH this project
  has asked CMake to find a package configuration file provided by
  "Sanitizers", but CMake did not find one.

  Could not find a package configuration file provided by "Sanitizers" with
  any of the following names:

    SanitizersConfig.cmake
    sanitizers-config.cmake

  Add the installation prefix of "Sanitizers" to CMAKE_PREFIX_PATH or set
  "Sanitizers_DIR" to a directory containing one of the above files.  If
  "Sanitizers" provides a separate development package or SDK, be sure it has
  been installed.


-- Installing auxiliary files in path: lib/appimagekit
CMake Error at src/CMakeLists.txt:176 (add_sanitizers):
  Unknown CMake command "add_sanitizers".


-- Configuring incomplete, errors occurred!
```

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -1,5 +1,12 @@
 find_package(Sanitizers)
 
+if(NOT COMMAND add_sanitizers)
+    message(WARNING "Sanitizers CMake module not found, sanitizer support is not available")
+    # defining a no-op function to avoid CMake errors
+    function(add_sanitizers ...)
+    endfunction()
+endif()
+
 set(APPIMAGEKIT_RUNTIME_ENABLE_SETPROCTITLE OFF CACHE BOOL "Useful for $TARGET_APPIMAGE; see issue #763")
 
 # set up build script

